### PR TITLE
boostrap miniconda directly

### PR DESCRIPTION
**Pull request by jhoblitt**
Instead of installing it as an EUPS distrib product.  This removes the
need to have a pre-existing python environment from which to get EUPS
running.  It also ensures that if miniconda is installed, it will be the
interpreter used by EUPS.  # Please enter the commit message for your
changes. Lines starting

### CODE REVIEW

**Comment by jhoblitt** [comment]
As a side comment, this script could use overall some style cleanup / refactoring but I've tried to keep the number of non-functional changes down in order to not pollute the diff.

File: scripts/newinstall.sh [hunk 1]
```diff
@@ -253,30 +354,62 @@ else:
...
+		miniconda::lsst_env "${PYTHON_VERSION}" "${LSSTSW_REF}"
+
+		CMD_SETUP_MINICONDA_SH="export PATH=\"${miniconda_path}/bin:\${PATH}\""
+		CMD_SETUP_MINICONDA_CSH="setenv PATH ${miniconda_path}/bin:\$PATH)"
+	fi
+fi
+
+# By default we use the PATH Python to bootstrap EUPS.
+# Set $PYTHON to override this or use the -P command line option.
+# $PYTHON is used to install and run EUPS and will not necessarily
```
Comment (timj) [inline scripts/newinstall.sh]: The `-P` option still sets `$PYTHON` but nothing uses it. Should `-P` now be setting `$EUPS_PYTHON`?

File: scripts/newinstall.sh [hunk 2]
```diff
@@ -47,44 +49,143 @@ else
...
+	miniconda_file_name="Miniconda${python_version}-${version}-${ana_platform}.sh"
+	echo "::: Deploying ${miniconda_file_name}"
+	$cmd $CURL -# -L -O "${miniconda_base_url}/${miniconda_file_name}"
+
+	$cmd bash "$miniconda_file_name" -b -p "$prefix"
+
+	if [[ $(uname -s) = Darwin* ]]; then
+		# run install_name_tool on all of the libpythonX.X.dylib dynamic
+		# libraries in miniconda
+		for entry in $prefix/lib/libpython*.dylib; do
```
Comment (timj) [inline scripts/newinstall.sh]: Do we still need this? It's not in `lsstsw` and I thought the modern conda python package got this right.

File: scripts/newinstall.sh [hunk 3]
```diff
@@ -33,8 +33,10 @@ EUPS_TARURL=${EUPS_TARURL:-"https://github.com/RobertLuptonTheGood/eups/archive/
 
 EUPS_PKGROOT=${EUPS_PKGROOT:-"http://sw.lsstcorp.org/eupspkg"}
 
-MINICONDA2_VERSION=${MINICONDA2_VERSION:-4.2.12.lsst2}
-MINICONDA3_VERSION=${MINICONDA3_VERSION:-4.2.12.lsst2}
+MINICONDA_VERSION=${MINICONDA_VERSION:-4.2.12}
+LSSTSW_REF=${LSSTSW_REF:-7c8e67}
```
Comment (timj) [inline scripts/newinstall.sh]: Can there be a comment here explaining that this ref is used to define the pinned package versions for conda? The main problem we have is that v13.0 of the stack wants different pinned versions to v12.0 and once you've installed a v13.0 stack your chances of building a v12.0 in the same tree are slim to none.

Comment (jhoblitt) [inline scripts/newinstall.sh]: OK

File: scripts/newinstall.sh (hunk 2 above)
Comment (jhoblitt) [inline scripts/newinstall.sh]: I don't know -- I was copying what the miniconda2/3 products were doing.  I can remove it to see if anything breaks.

File: scripts/newinstall.sh (hunk 1 above)
Comment (jhoblitt) [inline scripts/newinstall.sh]: Looks like something got lost in a rebase as I had universally fixed this...

**Pull request merged**